src/Components/WebAssembly/WebAssembly.Authentication/src/NavigationManagerExtensions.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+
+/// <summary>
+/// Extensions for <see cref="NavigationManager"/>.
+/// </summary>
+public static class NavigationManagerExtensions
```

Comment:
I suspect many people will try to call these extension methods and be surprised that we're requiring them to specify the URLs to the logout/login pages. It may feel like the framework should know those URLs and do the right thing for them.

Presumably that's not possible without adding some new feature, or you would have already done it. Is that right?

Reply:
It would require a DI service to be injected, since the URLs are configurable. I also don't want to do things that preclude in the future people having multiple authentication options enabled in the same app (like if you want to have AAD and AAD B2C). 

So being able to pass it in directly makes it more clear in the template and allows different options depending on the context it is being used on.

Reply:
For additional context, this is how we do it on the template right now:
https://github.com/dotnet/aspnetcore/blob/93d741edb0fa195f59d13743c35d21ee59099fc1/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Shared/LoginDisplay.IndividualLocalAuth.razor#L32

Reply:
Understood - thanks for clarifying. The template example looks totally reasonable.